icu4c/source/i18n/unicode/nounit.h

```diff
@@ -32,77 +32,43 @@ U_NAMESPACE_BEGIN
  * @see NumberFormatter
  * @draft ICU 60
  */
-class U_I18N_API NoUnit: public MeasureUnit {
-public:
+namespace NoUnit {
     /**
      * Returns an instance for the base unit (dimensionless and no scaling).
      *
-     * @return               a NoUnit instance
+     * Since ICU 68, this function returns the same value as the default MeasureUnit constructor.
+     *
+     * @return               a MeasureUnit instance
      * @draft ICU 60
      */
-    static NoUnit U_EXPORT2 base();
+    static inline MeasureUnit U_EXPORT2 base() {
```

Comment:
The class had U_I18N_API but the stand-alone functions only have U_EXPORT2. Sufficient? @jefgen

Reply:
I was looking for another non-class-member function and found this in unistr.h, outside of the UnicodeString class definition but still inside the icu namespace:
```
U_COMMON_API UnicodeString U_EXPORT2
operator+ (const UnicodeString &s1, const UnicodeString &s2);
```
So please add U_I18N_API to the front of these function definitions. Unsure exactly where, I guess between "inline" and the return type.

Reply:
Sorry, I missed this comment/ping before.

> The class had U_I18N_API but the stand-alone functions only have U_EXPORT2. Sufficient? @jefgen

FWIW, since this is now a "header-only" implementation I don't *think* the `U_I18N_API` macro is really needed for it.

The `U_I18N_API` macro marks things for dllexport, and before it was needed since the actual implementation for the functions was inside the .cpp file, so the translation unit would need them to be marked exported. The implementation is now only in the header, so anyone including the header already has access to the full implementation, and there's nothing to export.

For the UnicodeString `operator+`, it looks like implementation of the function is actually in the `unistr.cpp` file, so in that case it needs to be marked exported, which is likely why the header has `U_I18N_API` on it.

The `U_EXPORT2` macro only adds the `__cdecl` calling convention, but I'm less sure about that though.
Since this is header-only, I would think that it might not really matter much, as the consumers of the header would be calling the `MeasureUnit` functions directly.